### PR TITLE
Update discord game sdk to 3.2.1

### PR DESCRIPTION
**Pull request by Arthapz**
support linux
support macos

### CODE REVIEW

File: packages/d/discord/xmake.lua [hunk 1]
```diff
@@ -4,22 +4,43 @@ package("discord")
...
-            os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
-        end
 
         local configs = {}
+        if is_plat("windows") then
+            if package:is_arch("x64") then
+                os.cp("lib/x86_64/discord_game_sdk.dll", package:installdir("bin"))
+                os.cp("lib/x86_64/discord_game_sdk.dll.lib", package:installdir("lib"))
+                configs.linkflags = "-llib/x86_64/discord_game_sdk.dll"
```
Comment (waruqi) [inline packages/d/discord/xmake.lua:23]: no linkflags, `configs.linkdirs = "lib/x86_64"` and `configs.links = "discord_game_sdk.dll"`

File: packages/d/discord/xmake.lua [hunk 2]
```diff
@@ -4,22 +4,43 @@ package("discord")
...
+            else
+                os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
+                os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
+                configs.linkflags = "-llib/x86/discord_game_sdk.dll"
+            end
+        elseif is_plat("linux") then
+                os.cp("lib/x86_64/discord_game_sdk.so", package:installdir("lib"))
+                configs.linkflags = "-llib/x86_64/discord_game_sdk"
```
Comment (waruqi) [inline packages/d/discord/xmake.lua:31]: no linkflags, configs.linkdirs = "lib/x86_64" and configs.links = "discord_game_sdk"

File: packages/d/discord/xmake.lua [hunk 3]
```diff
@@ -4,22 +4,43 @@ package("discord")
...
-        if package:is_arch("x64") then
-            os.cp("lib/x86_64/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86_64/discord_game_sdk.dll.lib", package:installdir("lib"))
-        else
-            os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
-        end
 
         local configs = {}
+        if is_plat("windows") then
```
Comment (waruqi) [inline packages/d/discord/xmake.lua:19]: package:is_plat

File: packages/d/discord/xmake.lua [hunk 4]
```diff
@@ -4,22 +4,43 @@ package("discord")
...
+                configs.linkflags = "-llib/x86_64/discord_game_sdk.dll"
+            else
+                os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
+                os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
+                configs.linkflags = "-llib/x86/discord_game_sdk.dll"
+            end
+        elseif is_plat("linux") then
+                os.cp("lib/x86_64/discord_game_sdk.so", package:installdir("lib"))
+                configs.linkflags = "-llib/x86_64/discord_game_sdk"
+        elseif is_plat("macosx") then
```
Comment (waruqi) [inline packages/d/discord/xmake.lua:32]: package:is_plat

**Comment by Arthapz** [comment]
i don't understand why my watch is corrupted or the checksum is invalid

image

and when i used patch command by hand my patch is valid :/

**Comment by SirLynix** [comment]
It's probably because when you commit on git, it changes endlines from \n to \r\n (with the autoclrf option enabled).

Download the .patch from your PR and hash it.

**Comment by waruqi** [comment]
you can use `git diff > xxx.patch` to generate patch

File: packages/d/discord/xmake.lua [hunk 5]
```diff
@@ -4,26 +4,54 @@ package("discord")
...
+            local version = package:version()
+            if ((version:major() > 3) or (version:major() == 3 and version:minor() >= 2)) and package:is_arch("arm64") then
+                os.cp("lib/aarch64/discord_game_sdk.dylib", package:installdir("lib"))
+                configs.linkdirs = "lib/aarch64"
+
+            else
+                os.cp("lib/x86_64/discord_game_sdk.dylib", package:installdir("lib"))
+                configs.linkdirs = "lib/x86_64"
+            end
+            configs.links = "discord_game_sdk.dylib"
```
Comment (waruqi) [inline packages/d/discord/xmake.lua:49]: I don't understand why links are set to discordcpp when it is a static library.

Shouldn't `add_links("discordcpp", "discord_game_sdk")` or package:add("discordcpp", "discord_game_sdk") be configured in the package and then passed to the user project?

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake-repo/pull/1626

**Pull request closed without merging**